src/control/listIndexes.ts

```diff
@@ -1,28 +1,10 @@
 import { IndexOperationsApi } from '../pinecone-generated-ts-fetch';
-
-/**
- * A partial description of indexes in your project.
- *
- * For full information about each index, see
- * { @link Pinecone.describeIndex }
- */
-export type PartialIndexDescription = {
-  /** The name of the index */
-  name: string;
-};
-
-/** The list of indexes in your project */
-export type IndexList = Array<PartialIndexDescription>;
+import type { IndexMeta } from '../pinecone-generated-ts-fetch';
 
 export const listIndexes = (api: IndexOperationsApi) => {
-  return async (): Promise<IndexList> => {
-    const names = await api.listIndexes();
+  return async (): Promise<Array<IndexMeta>> => {
+    const response = await api.listIndexes();
 
-    // We know in a future version of the API that listing
-    // indexes should return more information than just the
-    // index names. Mapping these results into an object
-    // will allow us us to add more information in the future
-    // in a non-breaking way.
-    return names.map((n) => ({ name: n }));
+    return response.databases || [];
```

Comment:
I tried messing with the spec, but this `||` is mainly to handle the case where the response is an empty array. I could not get the OpenAPI spec to handle that and just return the empty array rather than `undefined`, so we basically do that here so we don't have to handle an optional response.

Open to feedback.

Reply:
Even if the spec generation stuff were easier to massage, it seems smart to be defensive like this.